lib/flags.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+class Flags {
+  static from(...args) {
+    const values = new Map(args.map((v, i) => [v, i]));
+    class FlagsClass {
+      static from(...args) {
+        return new FlagsClass(...args);
+      }
+
+      constructor(...args) {
+        this.value = 0;
```

Comment:
I think this implementation should use `Uint64` to store flags instead of the plain `number`; this will increase the total number of flags that can be held in a single type.